proto/teleport/quicpeering/v1alpha/dial.proto

```diff
@@ -0,0 +1,57 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+syntax = "proto3";
+
+package teleport.quicpeering.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+import "google/rpc/status.proto";
+
+option go_package = "github.com/gravitational/teleport/gen/proto/go/teleport/quicpeering/v1alpha;quicpeeringv1alpha";
+
+// Sent from the client to the server in a fresh QUIC stream to dial a server
+// through a QUIC proxy peering connection. The message is sent in protobuf
+// binary format, prefixed by its length encoded as a little endian uint32.
+message DialRequest {
+  string server_id = 1;
+  string conn_type = 2;
+
+  Addr source = 3;
+  Addr destination = 4;
+
+  // The time of the client, must be provided and within 5 minutes of the local
+  // server time for 0-RTT requests.
+  google.protobuf.Timestamp timestamp = 5;
+  // A random id for each dial request, must be provided and unique among
+  // requests presented in a 10-20 minute window for 0-RTT requests.
```

Comment:
That's an old value, it should read "at least 5 minutes". It's not an exact cutoff because having to keep track of exact "expiration" times is pretty silly since we're supposed to have 64 bits of entropy (which is not enough to guarantee uniqueness long-term, but is definitely enough for a much longer time than 5 or 10 minutes' worth of proxy peering connections), so the implementation I chose just swaps between two hashsets at the first request past the 5 minute cutoff time.

Reply:
Fixed in f804827f05